spec.md

```diff
@@ -1029,6 +1064,153 @@ It is NOT REQUIRED for a controller plugin to implement the `LIST_VOLUMES` capab
 
 ### Node Service RPC
 
+#### `NodeStageVolume`
+
+A Node Plugin MUST implement this RPC call if it has `STAGE_UNSTAGE_VOLUME` node capability.
+
+This RPC is called by the CO prior to the volume being consumed by any workloads on the node by `NodePublishVolume`.
+The Plugin SHALL assume that this RPC will be executed on the node where the volume will be used.
+This RPC SHOULD be called by the CO when a workload that wants to use the specified volume is placed (scheduled) on the specified node for the first time or for the first time since a `NodeUnstageVolume` call for the specified volume was called and returned success on that node.
+
+If the corresponding Controller Plugin has `PUBLISH_UNPUBLISH_VOLUME` controller capability and the Node Plugin has `STAGE_UNSTAGE_VOLUME` capability, then the CO MUST guarantee that this RPC is called after `ControllerPublishVolume` is called for the given volume on the given node and returns a success.
+The CO MUST guarantee that this RPC is called and returns a success before any `NodePublishVolume` is called for the given volume on the given node.
+
+This operation MUST be idempotent.
+If the volume corresponding to the `volume_id` is already staged to the `staging_target_path`, and is identical to the specified `volume_capability` the Plugin MUST reply `0 OK`.
```

Comment:
Can we clarify what to do when volume ID exists with a different staging path?

Reply:
Hi @cpuguy83,

Good catch. I firmly believe that CSI 0.2.0 needs to introduce extended error details because of this an other conditions where one of the several gRPC status codes just isn't enough information. For example, `CapacityError` for creating a volume with an invalid capacity range is an abuse of an eponymous code. It should be an `InvalidArgument` code with extended error details.

Reply:
As per the `NodeStageVolumeRequest` description: `The CO SHALL ensure uniqueness of staging_target_path per volume.`, so this is the CO's job to ensure. If the SP were to receive non-unique `staging_target_path` this would be undefined behavior, and I am ok leaving it as that.

Would be willing to define the behavior if you think that this is not enough for the spec and have a suggestion for what should happen.

Reply:
In this case, uniqueness of the path is not an issue, it's that the SP got a 2nd target path for for a given volume ID.

Reply:
I've clarified the wording to `The CO SHALL ensure that there is only one staging_target_path per volume.` hope that helps